kernel/src/process/process_vm/mod.rs

```diff
@@ -160,10 +163,24 @@ impl ProcessVm {
         &self.heap
     }
 
-    /// Clears existing mappings and then maps stack and heap vmo.
-    pub(super) fn clear_and_map(&self) {
+    /// Clears existing mappings and then maps the heap VMO to the current VMAR.
+    pub fn clear_and_map(&self) {
         let root_vmar = self.lock_root_vmar();
         root_vmar.get().clear().unwrap();
         self.heap.alloc_and_map_vm(&root_vmar.get()).unwrap();
     }
 }
+
+/// Renews the [`ProcessVm`] of the current process and then maps the heap VMO to the new VMAR.
+pub fn renew_and_map(ctx: &Context) {
+    let process_vm = ctx.process.vm();
+    let mut root_vmar = process_vm.lock_root_vmar();
+    let new_vmar = Vmar::<Full>::new_root();
+
+    let _guard = disable_preempt();
+    *ctx.thread_local.root_vmar().borrow_mut() = Some(new_vmar.dup().unwrap());
+    root_vmar.set_vmar(Some(new_vmar));
+    ctx.user_space().root_vmar().vm_space().activate();
```

Comment:
It seems better to call `new_vmar.vm_space().activate()` before `root_vmar.set_vmar(Some(new_vmar))`?